lib/src/rules/tighten_type_of_initializing_formals.dart

```diff
@@ -77,8 +77,7 @@ class _Visitor extends SimpleAstVisitor<void> {
             : e.leftOperand is NullLiteral
                 ? e.rightOperand
                 : null)
-        .where((e) => e != null)
-        .where((e) => context.typeSystem.isNullable(e!.staticType!))
+        .where((e) => e != null && context.typeSystem.isNullable(e.staticType!))
```

Comment:
I'll note that the `.whereType<Identifier>` would also guarantee that the remaining elements are not null, so you could eliminate the null check entirely by reversing (or combining) this line and the next.

Reply:
Good catch.  I also didn't see the dangling `!`.  Fixing.  Thank you!